certora/specs/Reverts.spec

```
@@ -176,21 +178,29 @@ rule setMaxRateRevertCondition(env e, uint256 newMaxRate) {
     assert lastReverted <=> e.msg.value != 0 || !callerIsAllocator || newMaxRate > maxMaxRate;
 }
 
-rule transferInputValidation(env e, address to, uint256 shares) {
+rule transferRevertCondition(env e, address to, uint256 shares) {
     bool toIsZeroAddress = to == 0;
     bool callerCanSendShares = canSendShares(e.msg.sender);
     bool toCanReceiveShares = canReceiveShares(to);
 
+    require shares + balanceOf(to) <= MAX_UINT256(); // To avoid overflow in the balanceOf check.
+    require shares <= balanceOf(e.msg.sender); // To avoid underflow in the balanceOf check.
+
     transfer@withrevert(e, to, shares);
-    assert toIsZeroAddress || !callerCanSendShares || !toCanReceiveShares => lastReverted;
+    assert toIsZeroAddress || !callerCanSendShares || !toCanReceiveShares || e.msg.value != 0 <=> lastReverted;
 }
 
-rule transferFromInputValidation(env e, address from, address to, uint256 shares) {
+rule transferFromRevertCondition(env e, address from, address to, uint256 shares) {
     bool fromIsZeroAddress = from == 0;
     bool toIsZeroAddress = to == 0;
     bool fromCanSendShares = canSendShares(from);
     bool toCanReceiveShares = canReceiveShares(to);
+    bool sharesLessEqualAllowance = e.msg.sender != from => (shares <= allowance(from, e.msg.sender)); // To avoid underflow in the allowance check.
```

let's try to be more high level in the comments
```suggestion
    bool sharesLessEqualAllowance = e.msg.sender != from => (shares <= allowance(from, e.msg.sender)); // msg.sender should have enough allowance.
```